zfs-replicate.sh

```diff
@@ -49,20 +47,37 @@ check_old_log() {
 }
 
 ## exit 0 and delete old log files
-exit_clean() {
+exit_clean () {
     ## print errors
     if [ "${1}x" != "x" ] && [ ${1} != 0 ]; then
         printf "Last operation returned error code: %s\n" "${1}"
     fi
     ## check log files
     check_old_log
+    clear_lock "${LOGBASE}"/.push.lock
+    clear_lock "${LOGBASE}"/.pull.lock
+    clear_lock "${LOGBASE}"/.snapshot.lock
     ## always exit 0
-    printf "Exiting...\n"
+    printf "SUCCESS\n"
+    exit 0 
+}
+
+exit_error () {
+    ## print errors
+    if [ "${1}x" != "x" ] && [ ${1} != 0 ]; then
+        printf "Last operation returned error code: %s\n" "${1}"
+    fi
+    ## check log files
+    check_old_log
+    clear_lock "${LOGBASE}"/.push.lock
+    clear_lock "${LOGBASE}"/.pull.lock
+    clear_lock "${LOGBASE}"/.snapshot.lock
+    printf "FAILED\n"
     exit 0
 }
```

Comment:
If adding a new function, shouldn't this one have a different exit code? The shared body between `exit_error` and `exit_clean` could be abstracted out into a function as well.

Reply:
The two function have the exact same code block, with the only exception being the “printf” message at the end.

I haven’t tried changing the exit 0 to exit 1 or exit 2 or anything. All I’m worried about at that point is the FAILED or SUCESS message.

doesn’t the exit 0 at the end refer to the function or either exit_error or exit_clean, meaning that the function completed successfully?

Reply:
Oh I see what you're saying. Yes they could be one single function, but how would you differentiate between exiting with a FAILED message vs a SUCCESS message?

Reply:
I’d leave it as is unless there is a further need to adapt the script. I’m going to resolve this for now.